topdown/eval.go

```diff
@@ -728,17 +728,19 @@ func (e *eval) evalCall(terms []*ast.Term, iter unifyIterator) error {
 	mock, mocked := e.functionMocks.Get(ref)
 	if mocked {
 		if m, ok := mock.Value.(ast.Ref); ok { // builtin or data function
-			mockCall := append([]*ast.Term{ast.NewTerm(m)}, terms[1:]...)
+			if _, ok := e.compiler.TypeEnv.Get(m).(*types.Function); ok {
```

Comment:
This check is new: if the replacement is a function, we call it. If it's not, we'll fall through to the replacement-by-value code path below (unchanged), and that'll do the right thing for refs to rules.